inference/models/yolov10/yolov10_object_detection.py

```diff
@@ -41,19 +51,48 @@ def predict(self, img_in: np.ndarray, **kwargs) -> Tuple[np.ndarray]:
             Tuple[np.ndarray]: NumPy array representing the predictions, including boxes, confidence scores, and class confidence scores.
         """
         predictions = self.onnx_session.run(None, {self.input_name: img_in})[0]
-        boxes = predictions[:, :, :4]
-        batch_size = boxes.shape[0]
-        num_boxes = boxes.shape[1]
-        class_confs = np.zeros(
-            (batch_size, num_boxes, self.num_classes), dtype=np.float32
-        )
-        for batch in range(batch_size):
-            for box in range(num_boxes):
-                class_id = int(predictions[batch, box, 5])
-                confidence = predictions[batch, box, 4]
-                class_confs[batch, box, class_id] = confidence
-
-        confs = predictions[:, :, 4:5]
-        predictions = np.concatenate([boxes, confs, class_confs], axis=2)
 
         return (predictions,)
+
+    def postprocess(
+        self,
+        predictions: Tuple[np.ndarray, ...],
+        preproc_return_metadata: PreprocessReturnMetadata,
+        confidence: float = DEFAULT_CONFIDENCE,
+        max_detections: int = DEFAUlT_MAX_DETECTIONS,
+        **kwargs,
+    ) -> List[ObjectDetectionInferenceResponse]:
+        """Postprocesses the object detection predictions.
+
+        Args:
+            predictions (np.ndarray): Raw predictions from the model.
+            img_dims (List[Tuple[int, int]]): Dimensions of the images.
+            confidence (float): Confidence threshold for filtering detections. Default is 0.5.
+            max_detections (int): Maximum number of final detections. Default is 300.
+
+        Returns:
+            List[ObjectDetectionInferenceResponse]: The post-processed predictions.
+        """
+        predictions = predictions[0]
+        predictions = np.append(predictions, predictions[..., 5:], axis=-1)
+        predictions[..., 5] = predictions[..., 4]
```

Comment:
@NickHerrig We are not very sure why the `max_class_confidence` need to be repeated twice in the `postprocess` of `ObjectDetectionBaseOnnxRoboflowInferenceModel`.

Reply:
yeah, side effect of some legacy decision made